File: Server/Models/AdditiveDosage.cs
#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
namespace Server.Models
{
    public class AdditiveDosage
    {
        public int Id { get; set; }
        public virtual Additive Additive { get; set; }
        public decimal Amount { get; set; }
        public string UnitofMeasure { get; set; }
    }
}
#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.

File: Client/src/models/AdditiveDosage.ts
import { Additive, AdditiveDto } from './Additive';

export default class AdditiveDosage {
    public id: number = 0;
    public additive: Additive = new Additive();
    public amount?: number;
    public unitofMeasure: string = "mL";
}

export class AdditiveDosageDto {
    public id?: number;
    public additive?: AdditiveDto;
    public amount?: number;
    public unitofMeasure?: string = "mL";
}
